From 59729bbcff38583d471ce3918be847e680a3514e Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dinuxbg@gmail.com>
Date: Sun, 16 Oct 2016 21:50:12 +0300
Subject: [PATCH 14/23] ld: testsuite: Sanitize output from ld

Apparently the leading dashes from an LD error message confuse DejaGnu.

This patch fixes the following DejaGnu error:

  --defsym:1: undefined symbol `foo2' referenced in expression
  failed with: <--defsym:1: undefined symbol `foo2' referenced in expression>, no expected output
  ERROR: tcl error sourcing /home/dinux/projects/misc/binutils-gdb/ld/testsuite/ld-gc/gc.exp.
  ERROR: usage: send [args] string
      while executing
  "send_log "$comp_output\n""
      (procedure "run_dump_test" line 376)
      invoked from within
  "run_dump_test "pr13683""
      (file "/home/dinux/projects/misc/binutils-gdb/ld/testsuite/ld-gc/gc.exp" line 136)
      invoked from within
  "source /home/dinux/projects/misc/binutils-gdb/ld/testsuite/ld-gc/gc.exp"
      ("uplevel" body line 1)
      invoked from within
  "uplevel #0 source /home/dinux/projects/misc/binutils-gdb/ld/testsuite/ld-gc/gc.exp"
      invoked from within
  "catch "uplevel #0 source $test_file_name""

Signed-off-by: Dimitar Dimitrov <dinuxbg@gmail.com>
---
 ld/testsuite/lib/ld-lib.exp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ld/testsuite/lib/ld-lib.exp b/ld/testsuite/lib/ld-lib.exp
index 9468d3e..4659243 100644
--- a/ld/testsuite/lib/ld-lib.exp
+++ b/ld/testsuite/lib/ld-lib.exp
@@ -966,6 +966,7 @@ proc run_dump_test { name {extra_options {}} } {
 	}
 
 	regsub "\n$" $comp_output "" comp_output
+	regsub "^--" $comp_output "" comp_output
 	if { $cmdret != 0 || $comp_output != "" || $check_ld(source) != "" } then {
 	    set exitstat "succeeded"
 	    if { $cmdret != 0 } { set exitstat "failed" }
-- 
2.9.3

